handlers/handlers.go

```diff
@@ -40,68 +39,60 @@ func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
 	w.WriteHeader(status)
 }
 
-func forwardFlorenceTokenIfRequired(req *http.Request) *http.Request {
-	if len(req.Header.Get(common.FlorenceHeaderKey)) > 0 {
-		ctx := common.SetFlorenceIdentity(req.Context(), req.Header.Get(common.FlorenceHeaderKey))
-		return req.WithContext(ctx)
-	}
-	return req
-}
-
-//GeographyRender ...
-func GeographyRender(rend RenderClient) http.HandlerFunc {
+//HomepageRender ...
+func HomepageRender(rend RenderClient, cli *codelist.Client) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
 		var page geographyHomepage.Page
 
-		resp, err := http.Get(`https://api.dev.cmd.onsdigital.co.uk/v1/code-lists` + localAuthority)
-		if err != nil {
-			setStatusCode(req, w, err)
-			return
-		}
-		b, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			setStatusCode(req, w, err)
-			return
-		}
-		var codelistresults models.CodeListResults
-		err = json.Unmarshal(b, &codelistresults)
+		codeListResults, err := cli.GetCodelistData()
 		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "error geting code-lists data for geography"), nil)
 			setStatusCode(req, w, err)
 			return
 		}
-		var codelist models.CodeList
-		err = json.Unmarshal(b, &codelist)
-		if err != nil {
-			setStatusCode(req, w, err)
-			return
-		}
-
-		geographyTypes := ""
-		for i := range codelistresults.Items {
-			log.Debug("test", log.Data{
-				"geographyTypesTest": codelistresults.Items[i],
-			})
-
-			geographyTypes = geographyTypes + codelistresults.Items[i].Name
-
-		}
 
-		page.Data.AreaTypes = []geographyHomepage.AreaType{
-			{Name: "Countries"},
-			{Name: "Regions"},
-			{Name: geographyTypes},
+		var types []geographyHomepage.Items
+		var wg sync.WaitGroup
+		var mutex = &sync.Mutex{}
+		for i := range codeListResults.Items {
+			wg.Add(1)
+			go func(codeListResults models.CodeListResults, i int, cli *codelist.Client) {
+
+				typesID := codeListResults.Items[i].Links.Self.ID
```

Comment:
When accessing an array, we need to check the length of it to make sure there is an element at that position before accessing it. If there were no items in the results right now, this would panic.

In this particular case though, it would actually be better to use one of the features of the `range` function in the `for` loop declaration. (https://gobyexample.com/range) Where currently we're only getting the index as we range over the items, we could get the value instead and assign that to a variable, and then use that variable inside the for loop.